src/Model/Endpoint.php

```diff
@@ -0,0 +1,747 @@
+     * ### Example:
+     *
+     * ```
+     * $id = 10;
+     * $article = $articles->get($id);
+     *
+     * $article = $articles->get($id, ['contain' => ['Comments]]);
+     * ```
+     *
+     * @param mixed $primaryKey primary key value to find
+     * @param array|\ArrayAccess $options options accepted by `Endpoint::find()`
+     * @throws \Cake\Datasource\Exception\RecordNotFoundException if the record with such id
+     * could not be found
+     * @return \Cake\Datasource\EntityInterface
+     * @see RepositoryInterface::find()
+     */
+    public function get($primaryKey, $options = [])
+    {
+        $key = (array)$this->primaryKey();
+        $alias = $this->alias();
+        foreach ($key as $index => $keyname) {
+            $key[$index] = $keyname;
+        }
+        $primaryKey = (array)$primaryKey;
+        if (count($key) !== count($primaryKey)) {
+            $primaryKey = $primaryKey ?: [null];
+            $primaryKey = array_map(function ($key) {
+                return var_export($key, true);
+            }, $primaryKey);
+
+            throw new InvalidPrimaryKeyException(sprintf(
+                'Record not found in endpoint "%s" with primary key [%s]',
+                $this->webservice(),
+                implode($primaryKey, ', ')
+            ));
+        }
+        $conditions = array_combine($key, $primaryKey);
+
+        $cacheConfig = isset($options['cache']) ? $options['cache'] : false;
+        $cacheKey = isset($options['key']) ? $options['key'] : false;
+        $finder = isset($options['finder']) ? $options['finder'] : 'all';
+        unset($options['key'], $options['cache'], $options['finder']);
+
+        $query = $this->find($finder, $options)->conditions($conditions);
+
+        if ($cacheConfig) {
+            if (!$cacheKey) {
+                $cacheKey = sprintf(
+                    "get:%s.%s%s",
+                    $this->connection()->configName(),
+                    $this->webservice(),
+                    json_encode($primaryKey)
+                );
+            }
+            $query->cache($cacheKey, $cacheConfig);
+        }
+
+        return $query->firstOrFail();
+    }
+
+    /**
+     * Creates a new Query instance for this repository
+     *
+     * @return Query
+     */
+    public function query()
+    {
+        return new Query($this->webservice(), $this);
+    }
+
+    /**
+     * Update all matching records.
+     *
+     * Sets the $fields to the provided values based on $conditions.
+     * This method will *not* trigger beforeSave/afterSave events. If you need those
+     * first load a collection of records and update them.
+     *
+     * @param array $fields A hash of field => new value.
+     * @param mixed $conditions Conditions to be used, accepts anything Query::where()
+     * can take.
+     * @return int Count Returns the affected rows.
+     */
+    public function updateAll($fields, $conditions)
+    {
+        // TODO: Implement updateAll() method.
```

Comment:
Yes I do actually, I'm going to use the ```Query```'s ```action``` method for this. I'll try to implement this tonight.